### ISSUE TITLE
Donation modals for bumble petition appear untranslated

### ISSUE THREAD

**Issue by data-sync-user** [MozillaFoundation/foundation.mozilla.org#12584]
Bug reported by Theo here: [https://mozilla-hub.atlassian.net/browse/MOFOTP-237](https://mozilla-hub.atlassian.net/browse/MOFOTP-237|smart-link)

Hi,

I’ve noticed the donation modals for the Bumble petition launched today appeared untranslated.

Here’s the French donation modal appearing after signing the petition:
[https://foundation.mozilla.org/fr/campaigns/bumble/?utm_medium=email&utm_source=mofo&utm_campaign=24-Bumble-Petition&utm_content=Text_Button_Sign-now&utm_term=fr&thank_you=true&c_id=701Pk00000IkbOpIAJ](https://foundation.mozilla.org/fr/campaigns/bumble/?utm_medium=email&utm_source=mofo&utm_campaign=24-Bumble-Petition&utm_content=Text_Button_Sign-now&utm_term=fr&thank_you=true&c_id=701Pk00000IkbOpIAJ|smart-link)

This modal is currently translated here: [https://foundation.mozilla.org/cms/snippets/wagtailpages/donationmodal/edit/171/](https://foundation.mozilla.org/cms/snippets/wagtailpages/donationmodal/edit/171/)

Note that old modals are not affected, only newly created modals.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/TP1-915)
┆Attachments: <a href="https://mozilla-hub.atlassian.net/rest/api/2/attachment/content/66684">image-20240722-165006.png</a> | <a href="https://mozilla-hub.atlassian.net/rest/api/2/attachment/content/66685">image-20240722-165429.png</a>

**Comment by AdalbertoMoz** [comment]
This has been fixed for the production website in the following languages:
- French
- German
- Spanish
- Polish
- Portuguese

For all other languages, the default English page is displayed. @TheoChevalier @simont-cr 

NOTE: If you visit the mentioned [French Donation Modal](https://foundation.mozilla.org/fr/campaigns/bumble/?utm_medium=email&utm_source=mofo&utm_campaign=24-Bumble-Petition&utm_content=Text_Button_Sign-now&utm_term=fr&thank_you=true&c_id=701Pk00000IkbOpIAJ%7Csmart-link), you can see the following:

![image](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/169944061/d312a919-0eb9-4288-a01c-06f88e600997)

For a more detailed description of the solution, please read the comment below.

**Comment by AdalbertoMoz** [comment]
After investigating in the CMS, I found the exact `CTA Donation Modal` used in this `Bumble Campaign Page`. If you click on the information button (red), you will see two important details:
1. The `Donation Modal` is available in 6 languages (green).
2. The **English** `Donation Modal` is used 9 times (blue).

![image](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/169944061/97216c82-82ef-465a-a235-672bcb2955a2)

To explain further, this means that:
1. The `Donation Modal` has been translated into 6 languages.
2. The **English** `Donation Modal` is being used for 9 languages, meaning it is used for all languages, and the other translated versions are not being used.

---

If you click on the "Usage" link (blue in the previous image), you can see that, as confirmed in the second point, the translated pages still redirect to the **English** `Donation Modal`:

![image](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/169944061/fc5227a9-c6aa-43de-89cd-e002738c6548)

And if you visit any of these pages inside the red rectangle of the last image, you can see the following warning:

![image](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/169944061/e310861b-8e0e-4b32-80af-f39043f7428f)

This warning indicates that we need to click on "Sync translated pages" for this specific `Bumble Campaign Page`. To do this, I found the original **English** `Bumble Campaign Page` and clicked the following button:

![image](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/169944061/42a64ad7-a915-4903-b42d-07157acc72fa)

After that, all the translated versions of the `Bumble Campaign Page` will be updated with a new field named "Donation Modals". The last step is to visit every translated version of the `Bumble Campaign Page` and click on "Publish". This will link the translated `Bumble Campaign Page` to the translated `Donation Modal` if available; otherwise, it will still use the **English** `Donation Modal`:

![image](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/169944061/b95010d5-d710-46a5-a591-7239b84daf3b)


---

And that's how I fixed this error. I know this may not be the ideal solution, but it worked for now. That's why I want to ask you, @danielfmiranda, since you have more experience with Wagtail and translations, is this really a bug? Or can this be fixed in an easier way?

**Comment by data-sync-user** [comment]
➤ diego commented:

Simon Acosta Torres I see this bug was already fixed, I tested the modal and it seems to be translated properly according to the locale defined. I also confirmed is available in 6 locales (see image below). I followed the thread and the investigation made by Adal where he highligthed the unpublished locale versions and how the english modal was being used on all other locales. 

!image-20240722-165006.png|width=1083,height=536,alt="image-20240722-165006.png"!

What indentified besides the mentioned above, is the “thank you” message that regardless the locale set (see image below), is keeping english, so I think can be defined in another task in case I’m not missing anything and this current task is completed and fixed. 

!image-20240722-165429.png|width=1138,height=586,alt="image-20240722-165429.png"!

Additionally I will double check with Theo to make sure it is all covered.
